src/components/Checkbox/UncontrolledCheckbox.tsx

```diff
@@ -0,0 +1,54 @@
+import React, {MutableRefObject, RefObject} from 'react';
+import clsx from 'clsx';
+import {useFocusRing} from '@react-aria/focus';
+
+import {AriaChecked, UncontrolledCheckboxProps} from './UncontrolledCheckbox.types';
+import './Checkbox.scss';
+
+export const UncontrolledCheckbox = React.forwardRef<HTMLInputElement, UncontrolledCheckboxProps>(
+    (props, ref) => {
+        const {
+            className,
+            indeterminate,
+            checked,
+            ...otherProps
+        } = props;
+        const defaultRef = React.useRef<HTMLInputElement>(null);
+        const resolvedRef = ref || defaultRef;
+        let ariaCheckedValue: AriaChecked = 'false';
+        const {isFocusVisible, focusProps} = useFocusRing();
+
+        if (indeterminate) {
+            ariaCheckedValue = 'mixed';
+        } else if (checked) {
+            ariaCheckedValue = 'true';
+        }
+
+        React.useEffect(() => {
+            (resolvedRef as MutableRefObject<HTMLInputElement>).current.indeterminate = indeterminate;
+        }, [resolvedRef, indeterminate]);
+
+        return (
+            <>
+                <div className={clsx('moonstone-checkbox', className)}>
+                    <input
+                        type="checkbox"
+                        className={clsx('moonstone-checkbox_input', {'hasFocus': isFocusVisible})}
+                        ref={resolvedRef}
+                        aria-checked={ariaCheckedValue}
+                        {...otherProps}
+                        {...focusProps}
+                    />
+                    <svg className={clsx('moonstone-checkbox_icon')} viewBox="0 0 21 21">
+                        { indeterminate
+                            ? <path d="M4.5 10.5L16.5 10.5" strokeLinecap="round"/>
+                            : <path d="M5 10.75L8.5 14.25L16 6" strokeLinecap="round"/>
+                        }
+                    </svg>
+                </div>
```

Comment:
Can we use the controlled version of the component as we did for the `TreeView` component to avoid duplicating code?

Reply:
Unfortunately,  the `react-table` library needs direct access to an uncontrolled version of the component to control. That's the only reason why I had to add this version.
Another approach we could take is to have the `Checkbox` component be controlled and the uncontrolled version only be accessed through a component called `UncontrolledCheckbox`. That way, only people who specifically want an uncontrolled version will use that component.

Reply:
The structure `Checkbox` component will stay as it is for now. [MOON-293](https://jira.jahia.org/browse/MOON-293) has been created to refactor this in the future.